arch/arc/kernel/Makefile

```diff
@@ -13,6 +13,22 @@ else
 obj-y	+= intc-arcv2.o
 endif
 
+# Currently ftrace only works on arcv3 due to ABI
+ifdef CONFIG_ISA_ARCV3
```

Comment:
I'd swap the order:
1. We do something for FTrace, so `#ifdef CONFIG_FTRACE`.
2. Inside we differentiate between ARCv3 and ARCv2/ARCompact.

Reply:
Agreed! Swapped the ifdefs